applicationset/services/pull_request/azure_devops.go

```diff
@@ -32,6 +35,29 @@ func (factory *devopsFactoryImpl) GetClient(ctx context.Context) (git.Client, er
 	return gitClient, nil
 }
 
+// devopsWorkloadIdentityFactoryImpl handles token refresh automatically for workload identity authentication
+type devopsWorkloadIdentityFactoryImpl struct {
```

Comment:
naming structs with a `XYZImpl` is not very go-ish 😄 
It was a java thing but we can remove it and let the struct be named `devopsWorkloadIdentityFactory` instead